recipes/default.rb

```diff
@@ -43,38 +46,67 @@
   path crt_file
   source node.loggly.tls.cert_url
   checksum node.loggly.tls.cert_checksum
-  notifies :restart, 'service[rsyslog]', :immediate
+  notifies :restart, 'service[rsyslog]', :delayed
 end
 
-# Set up tags
-tags = node.loggly.tags || []
-tags = tags.map { |tag| "tag=\\\"#{tag}\\\"" }.join(' ')
+# By default, use templates with configuration syntax for rsyslog versions
+# 7.x and higher.
+rsyslog_conf_source = 'rsyslog-loggly.conf.erb'
+files_conf_source = 'files.conf.erb'
+app_conf_source = 'apps.conf.erb'
+
+# If rsyslog version 6.x or below is installed use templates with the older syntax
+ruby_block 'set_version6_sources_if_needed' do
+  block do
+    if rsyslog_major_version <= 6
+      set_version6_source(node.loggly.rsyslog.conf, 'rsyslog-loggly-version6.conf.erb')
+      set_version6_source(node.loggly.rsyslog.files_conf, 'files-version6.conf.erb')
+      node.loggly.apps.keys.each do |app_name|
+        set_version6_source(app_conf(app_name), 'apps-version6.conf.erb')
```

Comment:
What do you think about adding something like version7 to the default template as well? I think we **may** support version 8 as well.